### PR TITLE
refactor CI and use `tox.ini` with Tox 4 and PEP517 build

### PR DESCRIPTION
**Pull request by zacharyburnett**
In an attempt to standardize CI across repos and make it less confusing, this PR reintroduces `tox` and uses the `toxenv` naming scheme from the Romancal repo.

image

Additionally, `toxenvs` in the `check` matrix run concurrently with other tests.

**Checklist**
- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [x] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#     tox -l -v
+#
+
+[testenv:check-style]
+description = check code style, e.g. with flake8
+skip_install = true
+deps =
+    flake8
+commands =
+    flake8 --count src tests {posargs}
```
Comment (braingram) [inline tox.ini:22]: Nice addition to include tests in the flake8 check.

**Review by braingram — APPROVED** [review]
Thanks for updating this. Looks good to me.

**Review by hbushouse — APPROVED** [review]

**Comment by braingram** [comment]
@zacharyburnett and/or @nden Is there someone with the required permissions that can update the branch protection rules to reflect the CI changes in this PR?
Specifically, I think the following checks were replaced (format "old -> new")
- Bandit security audit  -> CI / check-security
- Code style checks -> CI / check-style
- Coverage  -> CI / test-xdist-cov
- test (3.9, macos-latest) -> CI / test-xdist (Python 3.9, macos-latest)
- test (3.9, ubuntu-latest) -> CI / test-xdist (Python 3.9, ubuntu-latest)

An example PR with old style expectations and new CI is here: https://github.com/spacetelescope/stdatamodels/pull/110

**Comment by zacharyburnett** [comment]
> @zacharyburnett and/or @nden Is there someone with the required permissions that can update the branch protection rules to reflect the CI changes in this PR? Specifically, I think the following checks were replaced (format "old -> new")
> 
> * Bandit security audit  -> CI / check-security
> * Code style checks -> CI / check-style
> * Coverage  -> CI / test-xdist-cov
> * test (3.9, macos-latest) -> CI / test-xdist (Python 3.9, macos-latest)
> * test (3.9, ubuntu-latest) -> CI / test-xdist (Python 3.9, ubuntu-latest)
> 
> An example PR with old style expectations and new CI is here: #110

sure thing, changed in the settings

**Pull request merged**